source/WorldSetup.cc

```diff
@@ -34,22 +36,46 @@ void worldSetup(emp::Ptr<SymWorld> world, emp::Ptr<SymConfigBase> my_config) {
   double comp_host_1 = 0;
   double comp_host_2 = 0.95;
   //inject organisms
-  for (size_t i = 0; i < POP_SIZE; i++){
-    emp::Ptr<Host> new_org;
-    if (random_phen_host) {new_org.New(&random, world, my_config, random.GetDouble(-1, 1));
-    } else if (my_config->COMPETITION_MODE() && i%2==0) {
-        new_org.New(&random, world, my_config, comp_host_1);
-    } else if (my_config->COMPETITION_MODE() && i%2==1) {
-        new_org.New(&random, world, my_config, comp_host_2);
-    } else { new_org.New(&random, world, my_config, my_config->HOST_INT());
+
+  if(my_config->PGG()==0){
+     
+    for (size_t i = 0; i < POP_SIZE; i++){
+      emp::Ptr<Host> new_org;
+      
+      if (random_phen_host) {new_org.New(&random, world, my_config, random.GetDouble(-1, 1));
+      } else if (my_config->COMPETITION_MODE() && i%2==0) {
+          new_org.New(&random, world, my_config, comp_host_1);
+      } else if (my_config->COMPETITION_MODE() && i%2==1) {
+          new_org.New(&random, world, my_config, comp_host_2);
+      } else { new_org.New(&random, world, my_config, my_config->HOST_INT());
+      }
+      //Currently hacked because there isn't an AddOrg function, but there probably should be
+      if(my_config->GRID()) {
+        world->AddOrgAt(new_org, emp::WorldPosition(world->GetRandomCellID()));
+      } else {
+        world->AddOrgAt(new_org, world->size());
+      }
+      //world.Inject(*new_org);
     }
-    //Currently hacked because there isn't an AddOrg function, but there probably should be
-    if(my_config->GRID()) {
-      world->AddOrgAt(new_org, emp::WorldPosition(world->GetRandomCellID()));
-    } else {
-      world->AddOrgAt(new_org, world->size());
+  } else{
```

Comment:
This is a whole lot of repeated code, you could probably just have the emp::Ptr<Host> new org in an if statement instead from the looks of it.

Reply:
This looks like it will be easier to fix once #67 is resolved - should we hold off on fixing it until #67 is fixed?

Reply:
I think it should be easy in this case since it looks like the only line that is different between these blocks is " emp::Ptr<PggHost> new_org;" versus "emp::Ptr<Host> new_org;" so the MakeNew wouldn't be needed anyway

Reply:
Sorry if I'm being dumb here, but if he puts it in an if statement won't it go out of scope once the if statement ends?

Reply:
I tried to put the emp::Ptr new_org in an if statment before but it returns an error said that new_org was not decleared in the scope.